src/borg/testsuite/repository.py

```diff
@@ -714,6 +714,14 @@ def test_repair_missing_segment(self):
         self.check(repair=True, status=True)
         self.assert_equal(set([1, 2, 3]), self.list_objects())
 
+    def test_detect_attic_segment(self):
+        self.add_objects([[1, 2, 3]])
+        with open(os.path.join(self.tmppath, 'repository', 'data', '0', '1'), 'r+b') as fd:
+            fd.seek(0)
+            fd.write(ATTIC_MAGIC)
+        self.assert_raises(Repository.AtticRepository, lambda: self.get_objects(4))
+        self.assert_raises(Repository.AtticRepository, lambda: self.check())
```

Comment:
looks ok. another way would be to unpack the attic.tgz we have in the testsuite (which contains a repo made by attic).